pkg/controller/bundlec/bundle_sync_task.go

```diff
@@ -282,6 +285,55 @@ func (st *bundleSyncTask) deleteRemovedResources() (retriableError bool, e error
 			continue
 		}
 
+		deletionDelayAnnotation, ok := m.GetAnnotations()[smith.DeletionDelayAnnotation]
+		if ok {
+			// Trigger the "deletion delay" logic
+			deletionDelay, err := time.ParseDuration(deletionDelayAnnotation)
+			if err != nil {
+				return false, errors.Wrap(err, "failed to parse deletion delay duration")
```

Comment:
This should not stop the loop. Just log and continue to the next object. Also may emit an event on the bundle or on the object.

Reply:
Include the annotation name and value that failed to parse? Maybe that is too verbose... But it shouldn't happen often or at all so verbose is ok =)

Reply:
moving code to a separate function helped.